presto-main/src/main/java/com/facebook/presto/sql/relational/Signatures.java

```diff
@@ -155,9 +155,13 @@ public static Signature trySignature(Type returnType)
         return new Signature(TRY, SCALAR, returnType.getTypeSignature());
     }
 
-    public static Signature bindSignature(Type returnType, Type valueType, Type functionType)
+    public static Signature bindSignature(Type returnType, List<Type> argumentTypes)
```

Comment:
I think it makes more sense here to have `valueTypes` and `functionType` separate.